chainer/functions/hierarchical_softmax.py

```diff
@@ -184,14 +184,16 @@ def create_huffman_tree(word_counts):
         raise ValueError('Empty vocabulary')
 
     q = six.moves.queue.PriorityQueue()
-    for w, c in six.iteritems(word_counts):
-        q.put((c, w))
+    # Use unique id when two entries have the same count.
+    # Note that `iteritem` is randomly ordered.
+    for i, (w, c) in enumerate(six.iteritems(word_counts)):
```

Comment:
I think `id` instead of `i` is preferable because it make clear what `unique id` indicates